code/API_definitions/kyc-match.yaml

```
@@ -0,0 +1,536 @@
+                    addressMatch: 'true'
+                    streetNameMatch: 'true'
+                    streetNumberMatch: 'true'
+                    postalCodeMatch: 'true'
+                    regionMatch: 'true'
+                    localityMatch: 'not_available'
+                    countryMatch: 'true'
+                    houseNumberExtensionMatch: 'not_available'
+                    birthdateMatch: 'false'
+                    emailMatch: 'false'
+                    genderMatch: 'false'
+
+        '400':
+          $ref: '#/components/responses/Generic400'
+
+        '401':
+          $ref: '#/components/responses/Generic401'
+
+        '403':
+          $ref: '#/components/responses/Generic403'
+
+        '404':
+          $ref: '#/components/responses/Generic404'
+
+        '500':
+          $ref: '#/components/responses/Generic500'
+
+        '503':
+          $ref: '#/components/responses/Generic503'
+
+        '504':
+          $ref: '#/components/responses/Generic504'
+
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+
+    apiKey:
+      type: apiKey
+      in: header
+      name: X-KDDI-API-KEY
+      description: API key given to each client, to be used for API-GW authentication.
+    Authorization:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+      description: Access toke given by IdP(authentication)
+
+  schemas:
+    KYC_MatchRequestBody:
+      type: object
+      description: Payload to validate the customer data.
+
+      properties:
+        phoneNumber:
+          type: string
+          description: A phone number belonging to the user in E.164 format (starting with country code). Optionally prefixed with '+'.
+
+        idDocument:
+         type: string
+         description: Id number associated to the id_document. The value may also contain letters.
+
+        name:
+          type: string
+          description: Full name of the customer.
+
+        givenName:
+          type: string
+          description: First name/given name of the customer. It may include a compound first name or a second/middle name.
+
+        familyName:
+          type: string
+          description: Surname/family name of the customer. It may include a compound last name or an additional last name.
+
+        nameKanaHankaku:
+          type: string
+          description: Reading of the full name of the customer in full-width Kana format.
+
+        nameKanaZenkaku:
+          type: string
+          description: Reading of the full name of the customer in half-width Kana format.
+
+        middleNames:
+          type: string
+          description: Middle names of the customer
+
+        familyNameAtBirth:
+          type: string
+          description: Family name at birth of the customer
+
+        address:
+          type: string
+          description: Address of the customer

```

I think we should give a more detailed description here. For example, do we allow to input the type of street (Street, Avenue, Platz, etc? do we add the locality/region/country, etc?

```suggestion
          description: Address of the customer. This field can be the combination of multiple partial address parameters as streetName and streetNumber. It should not include street type.

```